libfreerdp/core/childsession.c

```diff
@@ -192,15 +357,29 @@ static int transport_bio_named_uninit(BIO* bio)
 static int transport_bio_named_new(BIO* bio)
 {
 	WINPR_ASSERT(bio);
-	BIO_set_flags(bio, BIO_FLAGS_SHOULD_RETRY);
 
 	WINPR_BIO_NAMED* ptr = (WINPR_BIO_NAMED*)calloc(1, sizeof(WINPR_BIO_NAMED));
 	if (!ptr)
 		return 0;
 
+	if (!ringbuffer_init(&ptr->readBuffer, 0xfffff))
+		goto error_ringbuffer;
+
+	ptr->readEvent = CreateEventA(NULL, TRUE, FALSE, NULL);
+	if (!ptr->readEvent || ptr->readEvent == INVALID_HANDLE_VALUE)
+		goto error_event;
+
+	ptr->readOverlapped.hEvent = ptr->readEvent;
+
 	BIO_set_data(bio, ptr);
 	BIO_set_flags(bio, BIO_FLAGS_SHOULD_RETRY);
 	return 1;
+
+error_event:
```

Comment:
shouldn´t this be cleaned up by `transport_bio_named_free` or `transport_bio_named_uninit`?

Reply:
I don't think that's possible as these functions rely on `BIO_set_data()` being called on the BIO (what we do at the end of transport_bio_named_new

Reply:
then it would be useful to wrap the `cleanup` in a separate function as `ptr = BIO_get_data(); cleanup(ptr); BIO_set_data(NULL);` is what is done?